src/pocketdb/SQLiteDatabase.cpp

```diff
@@ -110,6 +114,86 @@ namespace PocketDb
         }
     }
 
+    void SQLiteDatabase::InitMigration()
+    {
+
+        /*** Update Pocket DB from 0 to version 1 ********************************************
+         *
+         * This migration is necessary to fill the database with TxOutputs
+         * records with OP_RETURN records - these records were not saved initially in the database
+         * 
+         * It is also necessary to fill in the TxInputs table
+         * 
+         * Since the data of these tables become critically important for the operation of the node,
+         * it is necessary to make sure that they are available and valid.
+         * To do this, a full chainActive scan is performed, checking for the presence of all data.
+         * 
+         * Only after the end of the process, the DB version will be increased - in case of a process failure.
+         * 
+         ************************************************************************************/
+        const string dbNameMain = "main";
+        const int newDbVersion = 1;
+        int dbVersion = SystemRepoInst.GetDbVersion(dbNameMain);
+
+        if (dbVersion < newDbVersion)
+        {
+            int i = 0;
+            int percent = ChainActive().Height() / 100;
+            int64_t startTime = GetTimeMicros();
+
+            while (i <= ChainActive().Height() && !ShutdownRequested())
+            {
+                try
+                {
+                    // Read block from disk
+                    CBlock block;
+                    CBlockIndex* pblockindex = ChainActive()[i];
+                    if (!pblockindex || !ReadBlockFromDisk(block, pblockindex, Params().GetConsensus()))
+                    {
+                        LogPrintf("Failed read block %s from disk. Stopping\n", pblockindex->GetBlockHash().GetHex());
+                        StartShutdown();
+                        return;
+                    }
+
+                    for (const auto& tx : block.vtx)
+                    {
+                        // The default logic is used to deserialize the necessary data
+                        // We can use standard methods to write data to the database, because inside `INSERT OR IGNORE` is used
+                        // This will allow us to record only the missing data without changing the existing ones.
+                        if (auto[ok, ptx] = PocketServices::Serializer::DeserializeTransaction(tx); ok)
+                        {
+                            TransRepoInst.InsertTransactionInputs(ptx);
```

Comment:
Is there risk of creating duplicate records if the database has already been migrated?

Reply:
"INSERT OR IGNORE" should check the entry by key. In fact, I investigated the problem more deeply and realized that empty transactions and inputs are not critical to us - only empty outs are critical. I'll come back to this problem and maybe remove the migration logic, because it takes a very long time on an average PC. Perhaps it will be faster to download the snapshot.